.github/workflows/ci-test.yml

```diff
@@ -989,3 +989,94 @@ jobs:
           # or python docs `-p` on CI checks until we resolve the best way to
           # build them for deployment
           ./scripts/build_docs.sh -b
+
+  land-blocking-test-forge:
+    name: Run land blocking test in Forge
+    runs-on: self-hosted
+    needs: [prepare, build-images, need-base-images]
+    if: ${{ always() && needs.build-images.result=='success' && needs.prepare.outputs.test-land-blocking == 'true' }}
+    outputs:
+      lbt-result: ${{ steps.post-test-result.outputs.forge-result }}
+    timeout-minutes: 45
+    steps:
+      - uses: actions/checkout@v2.3.4
+        with:
+          # This ensures that the tip of the PR is checked out instead of the merge between the base ref and the tip
+          # On `push` this value will be empty and will "do-the-right-thing"
+          ref: ${{ github.event.pull_request.head.sha }}
+      - name: Launch forge test
+        # NOTE Remember to update PR comment payload if cti cmd is updated.
+        run: |
+          set +e
+          date
+          export FGI_OUTPUT_LOG=$(mktemp)
+          echo "FGI_OUTPUT_LOG=$FGI_OUTPUT_LOG" >> $GITHUB_ENV
+          cmd=""
+          if ${{ needs.need-base-images.result!='success' }}; then
+            cmd="./scripts/fgi/run --tag ${{ needs.build-images.outputs.head-tag }} --suite land_blocking"
+          else
+            cmd="./scripts/fgi/run -E SLACK_URL=${{ secrets.WEBHOOK_FORGE }} --tag ${{ needs.build-images.outputs.head-tag }} --base-image-tag ${{ needs.need-base-images.outputs.prev-tag }} --suite land_blocking_compat"
+          fi
+          eval $cmd
+          ret=$?
+          echo "fgi exit code: $ret"
+          echo "FGI_REPRO_CMD=$cmd" >> $GITHUB_ENV
+          echo "FGI_EXIT_CODE=$ret" >> $GITHUB_ENV
+      - name: Post test results on PR
+        id: post-test-result
+        uses: actions/github-script@v4.0.2
+        with:
+          github-token: ${{secrets.GITHUB_TOKEN}}
+          # set result to do comparison instead of fail workflow
+          script: |
+            res='success';
+            let env_vars = process.env;
+            if (parseInt(env_vars.FGI_EXIT_CODE) != 0) {
+              res='failed';
+            }
+            echo "::set-output name=forge-result::$(echo $res)";
```

Comment:
This section seems to not do what the step is titled, posting a comment on with the result on the PR. Is this because you want to wait to post the forge result to the PR till after we deprecate cluster test?

Reply:
yes, at this stage I think we'd better still rely on CT result as the source of truth, and Forge test result is only used to do comparison with CT to valid stability of Forge. Once Forge is turned on, we will post Forge result instead

Reply:
Might be worth a comment explaining this

Reply:
There are some comments above compare_lbt_signals job, I can add some more here as well to avoid misleading others